network-store-iidm-impl/src/main/java/com/powsybl/network/store/iidm/impl/ThreeWindingsTransformerImpl.java

```diff
@@ -405,17 +450,33 @@ public void remove() {
     @Override
     public <E extends Extension<ThreeWindingsTransformer>> Collection<E> getExtensions() {
         Collection<E> superExtensions = super.getExtensions();
-        Collection<E> result;
+        Collection<E> result = new ArrayList<>();
+        result.addAll(superExtensions);
+        E extension = createPhaseAngleClock();
         if (connectablePositionExtension != null) {
-            result = new ArrayList<>();
-            result.addAll(superExtensions);
             result.add((E) connectablePositionExtension);
+        } else if (extension != null) {
+            result.add(extension);
         } else {
             result = superExtensions;
         }
         return result;
     }
 
+    private <E extends Extension<ThreeWindingsTransformer>> E createPhaseAngleClock() {
+        E extension = null;
+        ThreeWindingsTransformerPhaseAngleClockAttributes phaseAngleClock = resource.getAttributes().getPhaseAngleClock();
+        if (phaseAngleClock != null) {
+            extension = (E) new ThreeWindingsTransformerPhaseAngleClockImpl(this, phaseAngleClock.getPhaseAngleClockLeg2(), phaseAngleClock.getPhaseAngleClockLeg3());
+        }
+        return extension;
+    }
+
+    public ThreeWindingsTransformerImpl initPhaseAngleClockAttributes(int phaseAngleClockLeg2, int phaseAngleClockLeg3) {
+        resource.getAttributes().setPhaseAngleClock(new ThreeWindingsTransformerPhaseAngleClockAttributes(phaseAngleClockLeg2, phaseAngleClockLeg3));
```

Comment:
updateResource is missing, I think it is the case in all the exception you habe implemented